django/db/backends/postgresql/base.py

```diff
@@ -149,6 +151,12 @@ def get_connection_params(self):
             raise ImproperlyConfigured(
                 "settings.DATABASES is improperly configured. "
                 "Please supply the NAME value.")
+        # PostgreSQL has a maximum length of 63 for identifiers.
+        if settings_dict['NAME'] is not None and len(settings_dict['NAME']) > 63:
+            raise NotSupportedError(
```

Comment:
Should this not raise a `ImproperlyConfigured` error, as it's down to a user configuration error? That's what we raise in the cases above this.

Reply:
I am not quite sure on this. I think that it is more like a ```NotSupportedError``` because there is no problem with the settings file as such, it's just the limitation with PostgreSQL. A name value is required in any case, and its absence should raise an ```ImproperlyConfigured``` error. Does this seem valid ?

Reply:
What about using the checks framework?

Reply:
There is a similar check for MySQL in the Checks Framework here: https://docs.djangoproject.com/en/2.0/ref/checks/#database
It does look better, should this be used ? @schinckel

Reply:
Using the checks framework would be a good thing, but I don't think it's possible here since it's only run after a successful database connection is set up, which an overly long name prevents, right?

Reply:
@adamchainz that is correct, a very long database-name will be truncated by PostgreSQL and you won't be able to connect.

Reply:
The name will be truncated upon creation of the database as well. So, a connection might be possible. But discovering the error after the database has been created should ideally clean-up the database already created, and if the 63-character long prefixes of two different databases is the same, you might connect to the wrong database. @schinckel @adamchainz

Reply:
Any suggestions @adamchainz ?

Reply:
Do I need to make any changes to L155 ?

Reply:
I don't think so